frontend/src/components/Pages/MissionPage.tsx

```diff
@@ -34,6 +34,20 @@ export function MissionPage() {
         }
     }, [])
 
+    useEffect(() => {
+        const id = setInterval(() => {
+            if (missionId) {
+                apiCaller.getMissionById(missionId).then((mission) => {
+                    setSelectedMission(mission)
+                    apiCaller.getVideoStreamsByRobotId(mission.robot.id).then((streams) => {
+                        setVideoStreams(streams)
+                    })
+                })
+            }
```

Comment:
This is duplicated code from the above `useEffect`. Is it possible to reuse something here?